README.md

```diff
@@ -89,11 +91,14 @@ MyComponent.propTypes = {
   // it as an enum.
   optionalEnum: PropTypes.oneOf(['News', 'Photos']),
 
+  optionalElementOfType: PropTypes.elementOfType(['div', OtherComponent]),
```

Comment:
why allow it to take an array of types instead of a single one? `PropTypes.oneOfType` is the established way to compose multiple validators.

Separately, I think "oneOf" is always in the name when it's taking an array of multiples; thus, I think `elementOfType` taking one, or `oneOfElementType` taking an array, is what would make sense - and I think the latter sounds silly.

Reply:
The specific problem I'm trying to solve is the oneOfType scenario, but considering the need to whitelist more than one type of components:
```js
<CustomTable>
  <CustomHeader/>
  <CustomTableRow/>
  <CustomTableRow/>
  <CustomTableRow/>
  <CustomFooter/>
</CustomTable>

// this is what I'm looking for
CustomTable.propTypes = {
  children: elementsOfType([CustomHeader, CustomTableRow, CustomFooter])
}

or

<SVGSection>
  <use/>
  <g></g>
  <rectangle/>
  <CustomSvgLine />
</SVGSection>

SVGSection.propTypes = {
  children: elementsOfType([ ...SVG_CHILDREN, ...MY_CUSTOM_SVG_CHILDREN ])
}
```

I'm hoping to capture these use cases so that when allow reusability of component, I'm always covered by this type checker.

Reply:
`PropTypes.oneOfType([ ...SVG_CHILDREN, ...MY_CUSTOM_SVG_CHILDREN ].map(x => PropType.elementOfType(x)))` works just fine, without overcomplicating the propType validator.

Reply:
@ljharb that sounds reasonable, I will make the change.

Reply:
@ljharb fyi, one feature we will lose from this change is a error message that says `expected one of type [div, span, CustomComponent]`, if we use `oneOfType`, we will only be seeing the generic message `Invalid prop supplied ...`

Reply:
I believe there’s a way oneOfType can read info off of the validators it receives to improve the error message, but I’m not sure what it is.

Reply:
@ljharb i wonder how that would work if each proptype returns a different error message.